library/src/main/java/me/grantland/widget/AutofitHelper.java

```diff
@@ -128,19 +131,49 @@ private static void autofit(TextView view, TextPaint paint, float minTextSize, f
                     displayMetrics);
         }
 
-        if (size < minTextSize) {
-            size = minTextSize;
+        if (mIsHeightFitting) {
+            int targetHeight = view.getHeight() - view.getPaddingTop() - view.getPaddingBottom();
+            if (targetHeight <= 0) {
```

Comment:
it seems like we could invert this conditional so that we don't have to duplicate minTextSize check and setTextSize calls